corehq/apps/case_search/management/commands/compare_case_properties.py

```diff
@@ -0,0 +1,85 @@
+from django.core.management.base import BaseCommand
+from corehq.apps.es.case_search import CaseSearchES
+import timeit
+
+
+def define_query(domain, props, operator, case_type=None):
+    script = {
+        "bool": {
+            "must": [
+                {
+                    "script": {
+                        "script": "def val = _source['case_properties'].find{ it -> it.key == '%s' }?.value;"
+                        "_source['case_properties'].find{ it -> it.key == '%s' }?.value %s val;"
+                        % (props[0], props[1], operator),
+                        "lang": "groovy"
+                    }
+                }
+            ]
+        }
+    }
+    if case_type:
+        query_obj = (CaseSearchES()
+            .domain(domain)
+            .set_query(script))
+    else:
+        query_obj = (CaseSearchES()
+            .domain(domain)
+            .case_type(case_type)
+            .set_query(script))
+    return query_obj
+
+
+class Command(BaseCommand):
+    help = "Run test queries for perfomance benchmarks for comparison script queries."
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+        parser.add_argument('-ct', '--type', dest='type', type=str)
+        parser.add_argument('-cp', '--properties', dest='props', nargs='+', type=str)
+
+        # operator arguments, only one of these is permitted
+        parser.add_argument('-eq', dest='equals', action='store_true')
+        parser.add_argument('-neq', dest='not_equals', action='store_true')
+        parser.add_argument('-gt', dest='greater_than', action='store_true')
+        parser.add_argument('-lt', dest='less_than', action='store_true')
+        parser.add_argument('-gteq', dest='greater_equal', action='store_true')
+        parser.add_argument('-lteq', dest='less_equal', action='store_true')
```

Comment:
This is an interesting approach. Since an "operator" is required I would have made that a required arg that did a similar thing with the dictionary in lines 56-62, but threw an error if it wasn't one of those 6 values. I think this is still not ideal, but this way it's clear to the person running that it is required rather than optional and that looking at it it's clear only one is allowed. Did you consider this approach and land on this approach for a reason?

Reply:
ah add ups cool